src/App.jsx

```diff
@@ -14,45 +16,123 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABETIC = 'abc';
+const SORT_FIELD_LENGTH = 'length';
+
+function getFilteredGoods(goods, { sortField, isReverse }) {
+  let filteredGoods = [...goods];
+
+  if (filteredGoods) {
+    filteredGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case 'abc':
+          return good1.localeCompare(good2);
+        case 'length':
+          return good1.length - good2.length;
+        default:
+      }
+
+      return 0;
+    });
+  }
+
+  if (isReverse) {
+    filteredGoods = filteredGoods.reverse();
+  }
+
+  return filteredGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+  const [reset, setReset] = useState(false);
+  const hadResetButton = !isReverse || sortField;
+
+  const goods = getFilteredGoods(goodsFromServer, { sortField, isReverse });
+
+  const handleSortAlphabetically = () => {
+    setSortField(SORT_FIELD_ALPHABETIC);
+    setReset(true);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const handleSortByLength = () => {
+    setSortField(SORT_FIELD_LENGTH);
+    setReset(true);
+  };
+
+  const handleReset = () => {
+    setSortField('');
+    setIsReverse(false);
+    setReset(false);
+  };
+
+  const handleReverse = () => {
+    setReset(hadResetButton);
+    setIsReverse(!isReverse);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={handleSortAlphabetically}
+          type="button"
+          className={classNames(
+            `button is-info ${
+              sortField === SORT_FIELD_ALPHABETIC ? '' : 'is-light'
+            }`,
+          )}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={handleSortByLength}
+          type="button"
+          className={classNames(
+            `button is-success ${
+              sortField === SORT_FIELD_LENGTH ? '' : 'is-light'
+            }`,
```

Comment:
classNames utility should be used in other way
https://www.npmjs.com/package/classnames
```suggestion
          className={classNames(
            'button is-success',
            { 'is-light': sortField === SORT_FIELD_LENGTH }
```

Reply:
fix everywhere please